src/borg/archive.py

```diff
@@ -81,15 +84,36 @@ def __add__(self, other):
         stats.osize_parts = self.osize_parts + other.osize_parts
         stats.usize_parts = self.usize_parts + other.usize_parts
         stats.nfiles_parts = self.nfiles_parts + other.nfiles_parts
+        stats.chunking_time = self.chunking_time + other.chunking_time
+        stats.hashing_time = self.hashing_time + other.hashing_time
+        stats.files_stats["A"] = self.files_stats["A"] + other.files_stats["A"]
+        stats.files_stats["U"] = self.files_stats["U"] + other.files_stats["U"]
+        stats.files_stats["M"] = self.files_stats["M"] + other.files_stats["M"]
+        stats.files_stats["E"] = self.files_stats["E"] + other.files_stats["E"]
```

Comment:
considering the code now automatically initialises values to 0 for arbitrary new keys due to the defaultdict, you could use a loop here to iterate over all keys.

guess "all" would need to be the union of the key sets of self.files_stats and other.files_stats.